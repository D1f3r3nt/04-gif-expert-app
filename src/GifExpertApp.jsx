import { useState } from 'react'
import { AddCategory, GifGrid } from './components'

export const GifExpertApp = () => {

    const [ categories, setCategories ] = useState(['One Punch'])

    const handleAddCategory = ( newCategory ) => {
        if (categories.includes(newCategory)) return;

        setCategories([newCategory, ...categories])
    }

    return (
        <>
            <h1>Gif Expert App</h1>

            <AddCategory onNewValue={ handleAddCategory } />

            {
                categories.map( cat => <GifGrid key={ cat } category={ cat } />)
            }
        </>
    )
}