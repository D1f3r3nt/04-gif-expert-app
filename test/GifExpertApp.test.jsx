import {render} from "@testing-library/react";
import {GifExpertApp} from "../src/GifExpertApp.jsx";

describe('Pruebas en <GifExpertApp />', () => {

    test('Debe hacer match con el es snapshot', () => {
        const { container } = render( <GifExpertApp /> )
        expect( container ).toMatchSnapshot()
    })
})