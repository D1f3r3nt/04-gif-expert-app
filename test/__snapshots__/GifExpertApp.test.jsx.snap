// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe hacer match con el es snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gift"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h1>
    Cargando...
  </h1>
  <div
    class="card-grid"
  />
</div>
`;
